### PR TITLE
add mitre-url nullPointerException validations

### PR DESCRIPTION
**Pull request by AvivCX**
### Description

Fix nullPointerException when cxflow trying to write mitreUrl, and property not exist

### References

gh issue #416

### Testing

manual

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentation is a Wiki Update, please indicate desired changes within PR MD Comment*
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

File: src/main/java/com/checkmarx/flow/custom/CsvIssueTracker.java [hunk 1]
```diff
@@ -142,7 +142,9 @@ public Issue createIssue(ScanResults.XIssue issue, ScanRequest request) throws M
                     value = issue.getCve();
                     break;
                 case "recommendation":
-                    value = String.format(flowProperties.getMitreUrl(), issue.getCwe());
+                    if (!ScanUtils.anyEmpty(flowProperties.getMitreUrl(), issue.getCwe())) {
+                        value = String.format(flowProperties.getMitreUrl(), issue.getCwe());
+                    }
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/custom/CsvIssueTracker.java:147]: all other switch cases have a log line.

File: src/main/java/com/checkmarx/flow/custom/GitLabSecurityDashboard.java [hunk 2]
```diff
@@ -77,14 +78,17 @@ public void complete(ScanRequest request, ScanResults results) throws MachinaExc
...
+        if (!ScanUtils.empty(flowProperties.getMitreUrl())) {
+            identifiers.add(
+                    Identifier.builder()
+                            .type("cwe")
+                            .name("CWE-".concat(issue.getCwe()))
+                            .value(issue.getCwe())
+                            .url(String.format(flowProperties.getMitreUrl(), issue.getCwe()))
+                            .build()
+            );
+        }else {log.info("mitre-url property is empty");}
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/custom/GitLabSecurityDashboard.java:90]: please fix indentation and style

File: src/main/java/com/checkmarx/flow/service/JiraService.java [hunk 3]
```diff
@@ -527,7 +527,7 @@ private void mapCustomFields(ScanRequest request, ScanResults.XIssue issue, Issu
                                 if (issue.getLink() != null && !issue.getLink().isEmpty()) {
                                     recommendation.append("Checkmarx Link: ").append(issue.getLink()).append(HTMLHelper.CRLF);
                                 }
-                                if (!ScanUtils.empty(issue.getCwe())) {
+                                if (!ScanUtils.anyEmpty(flowProperties.getMitreUrl(), issue.getCwe())) {
```
Comment (ofersk) [inline src/main/java/com/checkmarx/flow/service/JiraService.java:530]: what do we do if there is no Mitre? do we just ignore it?

**Review by kmcdon83 — APPROVED** [review]

File: src/main/java/com/checkmarx/flow/custom/CsvIssueTracker.java (hunk 1 above)
Comment (AvivCX) [reply src/main/java/com/checkmarx/flow/custom/CsvIssueTracker.java:148]: Added logs

File: src/main/java/com/checkmarx/flow/custom/GitLabSecurityDashboard.java (hunk 2 above)
Comment (AvivCX) [reply src/main/java/com/checkmarx/flow/custom/GitLabSecurityDashboard.java]: Fixed

File: src/main/java/com/checkmarx/flow/service/JiraService.java (hunk 3 above)
Comment (AvivCX) [reply src/main/java/com/checkmarx/flow/service/JiraService.java:531]: yes. it should be empty

**Pull request merged**